syntax = "proto3";
package stock;

message ProductStock {
    string sku = 1;
    int32 quantity = 2;
}

message ProductsStock {
    repeated ProductStock productStock = 1; 
}

message SingleProductStockRequest {
    string sku = 1;
}

message ProductsStockRequest {
    repeated string skus = 1;
}

service ProductStockService {
    rpc GetProductStockBySku (SingleProductStockRequest) returns (ProductStock);
    rpc GetProductsStockBySkus (ProductsStockRequest) returns (ProductsStock);
}

